src/uu/expand/src/expand.rs

```diff
@@ -89,23 +89,36 @@ fn tabstops_parse(s: &str) -> (RemainingMode, Vec<usize>) {
                 }
                 _ => {
                     // Parse a number from the byte sequence.
-                    let num = from_utf8(&bytes[i..]).unwrap().parse::<usize>().unwrap();
+                    let s = from_utf8(&bytes[i..]).unwrap();
+                    if let Ok(num) = s.parse::<usize>() {
+                        // Tab size must be positive.
+                        if num == 0 {
+                            crash!(1, "tab size cannot be 0");
+                        }
 
-                    // Tab size must be positive.
-                    if num == 0 {
-                        crash!(1, "tab size cannot be 0");
-                    }
+                        // Tab sizes must be ascending.
+                        if let Some(last_stop) = nums.last() {
+                            if *last_stop >= num {
+                                crash!(1, "tab sizes must be ascending");
```

Comment:
same for the test here

Reply:
There is already a test for this with `test_tabs_must_be_ascending()`.